maltoolbox/attackgraph/attackgraph.py

```diff
@@ -493,6 +333,176 @@ def attach_attackers(self) -> None:
 
             attacker.entry_points = list(attacker.reached_attack_steps)
 
+    def _follow_expr_chain(
```

Comment:
I think this method should be cached. There can be some quite expensive expressions to evaluate that may repeat in various other expressions. There is an issue with `@cache` that expects the method args to be hashable but lists and Model.assets are not. We can replace lists with tuples and the first issue is solved. About the second issue, once the class factory is gone this will be possible for us to tackle. Should we defer caching to the factory removal PR or try to tackle here?

Reply:
Let's defer it for now. This pull request is overly complicated as is.

Reply:
https://github.com/mal-lang/mal-toolbox/issues/96